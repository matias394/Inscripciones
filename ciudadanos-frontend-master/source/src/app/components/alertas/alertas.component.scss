.alert-div{
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.alert-text{
  color:black;
}

.alert-close{
  margin-bottom:10px;
}
