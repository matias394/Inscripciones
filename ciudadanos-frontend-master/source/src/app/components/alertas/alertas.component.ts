import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';

@Component({
  selector: 'app-alertas',
  template: `<div class="alert-wrapper">
    <div role="alert" [ngClass]="type" class="alert-div">
      <p class="alert-text">{{ message }}</p>
      <button
        *ngIf="showCloseButton"
        type="button"
        class="close alert-close"
        data-dismiss="alert"
        (click)="close()"
      >
        &times;
      </button>
    </div>
  </div>`,
  styleUrls: ['./alertas.component.scss'],
})
export class AlertasComponent implements OnInit {
  @Input() type: string = '';
  @Input() message: string = '';
  @Input() showCloseButton: boolean = false;
  @Output() onClose = new EventEmitter<Event>();

  constructor() {}

  close() {
    this.onClose.emit();
  }

  ngOnInit(): void {}
}
