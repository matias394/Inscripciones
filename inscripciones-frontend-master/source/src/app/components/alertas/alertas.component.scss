.alert-div {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: fit-content;
  gap: 10px;
}

.alert-text {
  color: black;
}

.alert-close {
  margin-bottom: 8px;
}
