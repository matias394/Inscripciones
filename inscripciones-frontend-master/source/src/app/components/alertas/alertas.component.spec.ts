import { ComponentFixture, TestBed } from '@angular/core/testing';
import { before } from 'underscore';

import { AlertasComponent } from './alertas.component';

const messageAlert = 'Hola! Esto es un mensaje';

describe('AlertasComponent', () => {
  let component: AlertasComponent;
  let fixture: ComponentFixture<AlertasComponent>;
  let compile: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [AlertasComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AlertasComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compile = fixture.nativeElement;
  });

  it('Componente creado', () => {
    expect(component).toBeTruthy();
  });

  test('AlertasComponent snapshot', () => {
    expect(compile).toMatchSnapshot();
  });

  test('Contenedor', () => {
    fixture.detectChanges();
    const alertContainer = compile.querySelector('.alert-div');
    expect(alertContainer.ELEMENT_NODE > 0).toBeTruthy();
  });

  test('Crear alerta por tipo success', () => {
    component.type = 'alert alert-success';
    fixture.detectChanges();
    const alertType = compile.querySelectorAll('.alert-success');
    expect(alertType.length).toBe(1);
  });

  test('Mostrar el mensaje', () => {
    component.message = messageAlert;
    fixture.detectChanges();
    const message = compile.querySelector('.alert-text');
    expect(message?.textContent).toContain(messageAlert);
  });

  test('Comprobar si el boton de cerrar se muestra', () => {
    component.showCloseButton = true;
    fixture.detectChanges();
    const boton = compile.querySelectorAll('button');
    expect(boton.length).toBe(1);
  });

  test('Comprobar si el boton de cerrar No se muestra', () => {
    component.showCloseButton = false;
    fixture.detectChanges();
    const boton = compile.querySelectorAll('button');
    expect(boton.length).toBe(0);
  });

  test('Emitir eventor close', () => {
    component.showCloseButton = true;
    fixture.detectChanges();
    jest.spyOn(component.onClose, 'emit');
    const botonClose = compile.querySelector('[data-dismiss=alert]');
    botonClose.dispatchEvent(new Event('click'));
    expect(component.onClose.emit).toHaveBeenCalled();
  });
});
