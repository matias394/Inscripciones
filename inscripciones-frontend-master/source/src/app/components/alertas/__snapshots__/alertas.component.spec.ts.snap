// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertasComponent AlertasComponent snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="alert-wrapper"
  >
    <div
      class="alert-div"
      role="alert"
    >
      <p
        class="alert-text"
      >
        
      </p>
      <span>
        Hola!
      </span>
      <button
        class="close alert-close"
        data-dismiss="alert"
        type="button"
      >
         Ã— 
      </button>
      
    </div>
  </div>
</div>
`;
